client/components/editor/teaching-elements/Assessment/ImageMultipleChoice.vue

```diff
@@ -0,0 +1,326 @@
+      let answers = cloneDeep(this.answers);
+      answers.push('');
+      this.update({ answers });
+    },
+    removeAnswer(answerIndex) {
+      let answers = cloneDeep(this.answers);
+      let correct = cloneDeep(this.correct);
+      let feedback = cloneDeep(this.feedback);
+
+      answers.splice(answerIndex, 1);
+      const index = correct.indexOf(answerIndex);
+      if (index !== -1) correct.splice(index, 1);
+
+      correct.forEach((it, i) => {
+        if (it >= answerIndex) correct[i] = it - 1;
+      });
+
+      if (feedback) {
+        range(answerIndex, answers.length).forEach(it => {
+          feedback[it] = feedback[it + 1];
+        });
+        delete feedback[answers.length];
+      }
+
+      this.fieldErrors.splice(answerIndex, 1);
+
+      this.update({ answers, correct, feedback });
+    },
+    hasQuestionText() {
+      let questionContent = this.assessment.question[0].data.content || '';
+      return questionContent !== '';
+    },
+    hasCorrectAnswers() {
+      return this.correct.length > 0;
+    },
+    validate() {
+      let error = {};
+      if (!this.hasQuestionText()) {
+        error = formAlert('The question field may not be empty.');
+      } else if (this.answers.length < 2) {
+        error = formAlert('Please make at least two answers available.');
+      } else if (!this.hasCorrectAnswers()) {
+        error = formAlert('There needs to be at least one correct answer.');
+      }
+      this.$emit('alert', error);
+    },
+    addErrorMessage(index, message) {
+      if (!this.fieldErrors[index]) {
+        this.fieldErrors[index] = [];
+      }
+      if (!this.fieldErrors[index].includes(message)) {
+        this.fieldErrors[index].push(message);
+      }
+    },
+    getErrorMessages(index) {
+      return this.fieldErrors[index] || [];
+    },
+    resetErrorMessages(index) {
+      this.fieldErrors[index] = [];
+    },
+    getUniqueIndex(index) {
+      return `imc_${Date.now()}_${index}`;
+    }
+  },
+  watch: {
+    assessment() {
+      this.validate();
+    },
+    isEditing() {
+      if (!this.isEditing) this.fieldErrors = [];
+    }
+  }
+};
+</script>
+
+<style lang="scss" scoped>
+$errorRed: #d9534f;
+
+h5 {
+  display: block;
+  margin: 30px 0 10px;
+  font-size: 18px;
+  text-align: left;
+}
+
+ul {
+  clear: both;
+  padding: 5px 0 0 10px;
+  list-style: none;
+
+  li {
+    position: relative;
+    margin: 20px 0;
+    padding-left: 40px;
+
+    .form-control {
+      padding-left: 10px;
+    }
+
+    input[type=checkbox] {
+      position: absolute;
+      top: 5px;
+      left: 0;
+    }
+  }
+
+  .mdi-close {
+    position: absolute;
+    right: 5px;
+    bottom: 5px;
+    padding: 5px;
+    color: #888;
+    cursor: pointer;
+
+    &:hover {
+      color: darken(#888, 20%);
+    }
+  }
+}
+
+.image {
+  &-container {
+    display: inline-block;
+    position: relative;
+    min-width: 128px;
+    max-width: 128px;
+    min-height: 128px;
+    max-height: 128px;
```

Comment:
Whole point of writing SCSS is avoid repeating same value multiple times in a row :bowtie: 
Also, what kind of pattern is this?